packages/contracts/test/voting/allowlist-voting-setup.ts

```diff
@@ -0,0 +1,66 @@
+import {expect} from 'chai';
+import {BigNumberish} from 'ethers';
+import {ethers} from 'hardhat';
+
+import {AllowlistVotingSetupV1} from '../../typechain';
+import {deployNewDAO} from '../test-utils/dao';
+
+enum Op {
+  Grant,
+  Revoke,
+  Freeze,
+  GrantWithOracle,
+}
+
+const abiCoder = ethers.utils.defaultAbiCoder;
+const AddressZero = ethers.constants.AddressZero;
+const EMPTY_DATA = '0x';
+const EMPTY_BYTES_32 = `0x${'00'.repeat(32)}`;
+
+// TODO 1. add type GRANT/REVOKE check in permissions
+// TODO 2. in order to detect encode abi for deploy/update, use deployABI/updateABI
```

Comment:
What will happen with those todos?

Reply:
will be tackled in the next PR, it has it's own task.